app/src/main/java/com/courseproject/tindar/ChatActivity.java

```diff
@@ -1,35 +1,178 @@
+        TindarMessage testMessage5 = new TindarMessage("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam maximus erat nisl, a scelerisque leo euismod nec. Ut dapibus auctor augue, quis tempor tellus tincidunt vel. Fusce ut odio mauris. Nam nec finibus enim. Duis et nisi tristique, luctus leo id, facilisis dolor. Phasellus ac auctor odio, non mollis magna. Suspendisse tortor ipsum, consectetur vitae metus et, accumsan luctus erat. Sed gravida, ipsum vel mattis maximus, orci arcu convallis nunc, sed sagittis metus felis ac sapien. Integer non pellentesque massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et dignissim erat. Integer at nibh ac mi ultricies pulvinar. Morbi nec arcu nisi. Duis eu ligula auctor, dictum tortor a, condimentum velit. Suspendisse potenti. Sed fermentum lobortis blandit.", new Timestamp(2023, 05, 25, 18, 0, 0, 0),"user2", "user1");
+
+        this.loadedMessages = new ArrayList<MessageModel>();
+        this.loadedMessages.add(testMessage1);
+        this.loadedMessages.add(testMessage2);
+        this.loadedMessages.add(testMessage3);
+        this.loadedMessages.add(testMessage3);
+        this.loadedMessages.add(testMessage3);
+        this.loadedMessages.add(testMessage3);
+        this.loadedMessages.add(testMessage3);
+        this.loadedMessages.add(testMessage4);
+        this.loadedMessages.add(testMessage5);
+    }
+
+    // TODO: this should be refactored.
+    //      both to make future implementation of other message types easier,
+    //      and because I feel like the sending and creation of the new message should go elsewhere.
+    //      I don't think we should be importing Tindar messages.
+    // TODO: timezone support?
+
+    // TODO: chatInput should clear once the button is pressed, so that people don't have to delete
+    //          the previous message to send a new one
+    // TODO: shouldn't send empty messages
+    /**
+     * Called when sendMessageButton is called. Creates the new message, and informs adapter
+     * that it needs to update the display.
+     *
+     * Currently just adds the message to loadedMessages.
+     * Will probably call some sort of conversationManager class and be slightly reimplemented
+     * with the database.
+     *
+     * @return void
+     */
+    public void sentMessage(View v){
+        String input = (this.chatInput.getText()).toString();
+        MessageModel newMessage = new TindarMessage(input, new Timestamp(System.currentTimeMillis()),
```

Comment:
I'm not sure you can use entity all the way in the frontend. You probably need another model in the use case to use it in the frontend

Reply:
this is the front end model. a different model will be used for backend stuff.